src/schema/validator.js

```diff
@@ -4,28 +4,191 @@
  * Copying and unauthorised use of this material strictly prohibited.
  */
 
-const Ajv = require('ajv');
-const EPCISDocument = require('./EPCISDocument.schema.json');
+import Ajv from 'ajv';
+import addFormats from 'ajv-formats';
+import * as fieldNames from '../entity/field-names';
+import definitions from './definitions.json';
+import validationMode from '../settings';
+import EPCISDocument from './EPCISDocument.schema.json';
+import ObjectEvent from './ObjectEvent.schema.json';
+import AggregationEvent from './AggregationEvent.schema.json';
+import TransformationEvent from './TransformationEvent.schema.json';
+import TransactionEvent from './TransactionEvent.schema.json';
+import AssociationEvent from './AssociationEvent.schema.json';
+import ExtendedEvent from './ExtendedEvent.schema.json';
 
-const ajv = new Ajv({ useDefaults: true });
+/**
+ * @typedef {object} ValidatorResult
+ * @property {boolean} success - true if the validator found no errors.
+ * @property {Array<string>} errors - List of errors.
+ */
+const successResult = { success: true, errors: [] };
+
+const ajv = addFormats(new Ajv({ useDefaults: true, strict: false }), { mode: validationMode });
+
+/**
+ * Load a schema and include 'definitions'.
+ *
+ * @param {object} schema - schema to be loaded.
+ * @returns {object} Loaded and enhanced schema.
+ */
+const loadSchema = ((schema) => ({ ...schema, definitions })
+);
 
 /** Available schemas */
 const validators = {
```

Comment:
export this one. It might be cool to use valigateAgainstSchema outside of the SDK. You'll need to export it in index.js too